### PR TITLE
[flutter_tools] More Linux version detail

### PR DESCRIPTION
**Pull request by zanderso**
## Description

This PR collects more information about the linux version for reporting to analytics and the doctor.

## Related Issues

https://github.com/flutter/flutter/issues/50575

## Tests

I added the following tests:

Tests in os_test.dart.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
LGTM, other than offline nit.

File: packages/flutter_tools/test/general.shard/base/os_test.dart [hunk 1]
```diff
@@ -88,6 +89,73 @@ void main() {
...
+      final FakePlatform platform = FakePlatform(
+        operatingSystem: 'linux',
+        operatingSystemVersion: 'Linux 5.2.17-amd64 #1 SMP Debian 5.2.17 (2019-10-21 > 2018)',
+      );
+      final OperatingSystemUtils utils = createOSUtils(platform);
+      expect(utils.name, 'Linux 5.2.17-amd64');
+    });
+
+    testWithoutContext('on Mac', () {
+      when(mockProcessManager.runSync(<String>['sw_vers', '-productName']))
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/base/os_test.dart]: I think the FakeProcessManager ends up being easier to understand and update for these sorts of mocks.

File: packages/flutter_tools/test/general.shard/base/os_test.dart [hunk 2]
```diff
@@ -88,6 +89,73 @@ void main() {
...
+      }
+    }, overrides: <Type, Generator>{
+      OperatingSystemUtils: () => OperatingSystemUtils(
+        fileSystem: globals.fs,
+        logger: globals.logger,
+        platform: globals.platform,
+        processManager: globals.processManager,
+      ),
+    });
+  });
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/base/os_test.dart]: maybe: skip: const LocalPlatform().isWindows? Not sure which is better.

Comment (zanderso) [reply packages/flutter_tools/test/general.shard/base/os_test.dart]: Done.

File: packages/flutter_tools/test/general.shard/base/os_test.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/test/general.shard/base/os_test.dart]: Done.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Pull request merged**